src/unaudited/ECDSAServiceManagerBase.sol

```diff
@@ -0,0 +1,240 @@
+    {
+        return _getRestakeableStrategies();
+    }
+
+    /// @inheritdoc IServiceManagerUI
+    function getOperatorRestakedStrategies(
+        address _operator
+    ) external view virtual returns (address[] memory) {
+        return _getOperatorRestakedStrategies(_operator);
+    }
+
+    /**
+     * @notice Forwards the call to update AVS metadata URI in the AVSDirectory contract.
+     * @dev This internal function is a proxy to the `updateAVSMetadataURI` function of the AVSDirectory contract.
+     * @param _metadataURI The new metadata URI to be set.
+     */
+    function _updateAVSMetadataURI(
+        string memory _metadataURI
+    ) internal virtual {
+        IAVSDirectory(avsDirectory).updateAVSMetadataURI(_metadataURI);
+    }
+
+    /**
+     * @notice Forwards the call to register an operator in the AVSDirectory contract.
+     * @dev This internal function is a proxy to the `registerOperatorToAVS` function of the AVSDirectory contract.
+     * @param operator The address of the operator to register.
+     * @param operatorSignature The signature, salt, and expiry details of the operator's registration.
+     */
+    function _registerOperatorToAVS(
+        address operator,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature
+    ) internal virtual {
+        IAVSDirectory(avsDirectory).registerOperatorToAVS(
+            operator,
+            operatorSignature
+        );
+    }
+
+    /**
+     * @notice Forwards the call to deregister an operator from the AVSDirectory contract.
+     * @dev This internal function is a proxy to the `deregisterOperatorFromAVS` function of the AVSDirectory contract.
+     * @param operator The address of the operator to deregister.
+     */
+    function _deregisterOperatorFromAVS(address operator) internal virtual {
+        IAVSDirectory(avsDirectory).deregisterOperatorFromAVS(operator);
+    }
+
+    /**
+     * @notice Processes a batch of range payments by transferring the specified amounts from the sender to this contract and then approving the PaymentCoordinator to use these amounts.
+     * @dev This function handles the transfer and approval of tokens necessary for range payments. It then delegates the actual payment logic to the PaymentCoordinator contract.
+     * @param rangePayments An array of `RangePayment` structs, each representing a payment for a specific range.
+     */
+    function _payForRange(
+        IPaymentCoordinator.RangePayment[] calldata rangePayments
+    ) internal virtual {
+        for (uint256 i = 0; i < rangePayments.length; ++i) {
+            rangePayments[i].token.transferFrom(
+                msg.sender,
+                address(this),
+                rangePayments[i].amount
+            );
+            rangePayments[i].token.approve(
+                paymentCoordinator,
+                rangePayments[i].amount
+            );
+        }
+
+        IPaymentCoordinator(paymentCoordinator).payForRange(rangePayments);
```

Comment:
only 'gotcha' I can think of is if there are duplicate tokens, this will revert at the end due to insufficient allowance.
I'm fine chalking that up as "operator error" though.